charlotte_core/src/arch/x86_64/mod.rs

```diff
@@ -231,4 +270,119 @@ impl Api {
         logln!("Contiguous frame allocation and deallocation test complete.");
         logln!("Physical Memory Manager test suite finished.");
     }
+
+    fn vmm_self_test() {
+        logln!("Beginning VMM Self Test...");
+        let cr3 = unsafe { asm_get_cr3() };
+        let mut pm = PageMap::from_cr3(cr3).expect("Failed to create PageMap from CR3 value.");
+        logln!("PageMap created from current CR3 value.");
+
+        logln!("Starting page mapping test...");
+        let frame = PHYSICAL_FRAME_ALLOCATOR
+            .lock()
+            .allocate()
+            .expect("Failed to allocate frame.");
+        let vaddr = match VirtualAddress::try_from(0xFFFF800000000000) {
+            Ok(vaddr) => vaddr,
+            Err(e) => {
+                panic!("Failed to create VirtualAddress: {:?}", e);
+            }
+        };
+        pm.map_page(
+            vaddr,
+            frame,
+            PteFlags::Write as u64 | PteFlags::Global as u64 | PteFlags::NoExecute as u64,
+        );
```

Comment:
Please ignore or include `let _ =`  to suppress the returned error. Ideally, as this is testing the VMM, handeling the error and writing out a log accordingly would be even better.

Reply:
Agree with wrench in this one.